beacon-chain/p2p/discovery.go

```diff
@@ -47,6 +52,88 @@ type quicProtocol uint16
 // quicProtocol is the "quic" key, which holds the QUIC port of the node.
 func (quicProtocol) ENRKey() string { return "quic" }
 
+type listenerWrapper struct {
+	*sync.RWMutex
+	listener        *discover.UDPv5
+	listenerCreator func() (*discover.UDPv5, error)
+}
+
+func NewListener(listenerCreator func() (*discover.UDPv5, error)) (*listenerWrapper, error) {
```

Comment:
exported func NewListener returns unexported type *p2p.listenerWrapper, which can be annoying to use.
(DeepSource)

Reply:
Fixed